src/sentry/notifications/api/serializers/notification_action_request.py

```diff
@@ -0,0 +1,380 @@
+            raise serializers.ValidationError(
+                f"Service type of '{service_provider}' requires having an active integration"
+            )
+
+        if integration and service_provider != integration.provider:
+            raise serializers.ValidationError(
+                {
+                    "integration_id": f"Integration of provider '{integration.provider}' does not match service type of '{service_provider}'"
+                }
+            )
+        self.integration = integration
+
+    def validate_sentry_app_and_service(self, data: NotificationActionInputData):
+        if (
+            data["service_type"] == ActionService.SENTRY_APP.value
+            and data.get("sentry_app_id") is None
+        ):
+            service_type = ActionService.get_name(data["service_type"])
+            raise serializers.ValidationError(
+                {
+                    "sentry_app_id": f"Service type of '{service_type}' requires providing a sentry app id"
+                }
+            )
+
+    def validate_with_registry(self, data: NotificationActionInputData):
+        registration = NotificationAction.get_registration(
+            trigger_type=data["trigger_type"],
+            service_type=data["service_type"],
+            target_type=data["target_type"],
+        )
+        if not registration:
+            raise serializers.ValidationError(
+                "Combination of trigger_type, service_type and target_type has not been registered."
+            )
+        registration.validate_action(data=data)
+
+    def validate_slack_channel(
+        self, data: NotificationActionInputData
+    ) -> NotificationActionInputData:
+        """
+        Validates that SPECIFIC targets for SLACK service has the following target data:
+            target_display: Slack channel name
+            target_identifier: Slack channel id (optional)
+        NOTE: Reaches out to via slack integration to verify channel
+        """
+        if (
+            data["service_type"] != ActionService.SLACK.value
+            or data["target_type"] != ActionTarget.SPECIFIC.value
+        ):
+            return data
+
+        channel_name = data.get("target_display")
+        channel_id = data.get("target_identifier")
+        if not channel_name:
+            raise serializers.ValidationError(
+                {"target_display": "Did not receive a slack user or channel name."}
+            )
+
+        # If we've received a channel and id, verify them against one another
+        if channel_name and channel_id:
+            try:
+                validate_channel_id(
+                    name=channel_name,
+                    integration_id=self.integration.id,
+                    input_channel_id=channel_id,
+                )
+            except Exception as e:
+                # validate_channel_id raises user friendly validation errors!
+                raise serializers.ValidationError({"target_display": str(e)})
```

Comment:
After this move merges we should narrow which kinds of exceptions are caught here, as catching all exception types could lead to issues like the bot mentioned.